modules/user/src/main/DataForm.scala

```diff
@@ -15,6 +15,15 @@ final class DataForm(authenticator: Authenticator) {
 
   case class NoteData(text: String, mod: Boolean)
 
+  def username(user: User): Form[UserNameChange] = Form(mapping(
+    "userName" -> text.verifying("usernameNotSame", name =>
+      name.toLowerCase == user.username.toLowerCase && name != user.username)
+  )(UserNameChange.apply)(UserNameChange.unapply))
```

Comment:
no need for `mapping` here, you can use a `single` form and map directly to a `String`.

e.g. https://github.com/ornicar/lila/blob/master/modules/security/src/main/DataForm.scala#L140-L146